src/app/machines/views/MachineList/MachineListTable/AllCheckbox/AllCheckbox.test.tsx

```diff
@@ -0,0 +1,96 @@
+import { screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import configureStore from "redux-mock-store";
+
+import AllCheckbox from "./AllCheckbox";
+
+import { actions as machineActions } from "app/store/machine";
+import type { RootState } from "app/store/root/types";
+import {
+  rootState as rootStateFactory,
+  machineStateList as machineStateListFactory,
+  machineState as machineStateFactory,
+} from "testing/factories";
+import { renderWithMockStore } from "testing/utils";
+
+const mockStore = configureStore<RootState, {}>();
+
+let state: RootState;
+const callId = "123456";
+
+beforeEach(() => {
+  state = rootStateFactory({
+    machine: machineStateFactory({
+      lists: {
+        [callId]: machineStateListFactory(),
+      },
+    }),
+  });
+});
+
+it("is unchecked if there are no filters, groups or items selected", () => {
+  state.machine.selectedMachines = null;
+  renderWithMockStore(<AllCheckbox callId={callId} />, { state });
+  expect(screen.getByRole("checkbox")).not.toBeChecked();
```

Comment:
Might be worth testing in combination with a label (e.g. `expect(screen.getByRole("checkbox", { name: /all/ })).not.toBeChecked();`). This checkbox currently has none:

<img width="558" alt="image" src="https://user-images.githubusercontent.com/7452681/187371129-df512572-4e7d-4e13-8a0b-aa0d6d740c0e.png">

Reply:
Done!